safe_video/ui/helper_classes.py

```diff
@@ -5,53 +5,92 @@
 from .components import ModelTile
-from safe_video.number_plate_recognition import ObjectDetection, Censor, apply_censorship
+from safe_video.number_plate_recognition import ObjectDetection, Censor, apply_censorship, merge_results_list
 from ultralytics.engine.results import Results
 
 class ModelManager():
     def __init__(self, bounding_box_func):
         self.detection: ObjectDetection = ObjectDetection()
-        self.cls = self.detection.get_classes()[0:2]
-        self.active: dict[str, bool] = {c: True for c in self.cls}
+        self.cls_file_path = 'safe_video/upload_cache/cls_file.pkl'
+        self.cls = {}
+        if os.path.isfile(self.cls_file_path):
+            with open(self.cls_file_path, 'rb') as file:
+                self.cls = pickle.load(file)
+        self.active: dict[str, bool] = {c: True for c in self.cls.keys()}
         self.results: dict[str, dict[str, Results]] = dict() # dict[cls_id][img_id]
         self.bounding_box_func = bounding_box_func
 
     def toggle_active(self, cls_id):
         self.active[cls_id] = not self.active[cls_id]
 
+    def get_possible_cls(self):
+        return self.detection.get_classes()
+
+    def insert_new_cls(self, name, classes, active = True):
+        counter = 1
+        id = name
+        while id in self.cls.keys():
+            id = name + ' ' + str(counter)
+            counter +=1
+        self.cls[id] = classes
+        self.active[id] = active
+        self.update_cls_file()
+        return id
+
+    def edit_cls(self, old_name, new_name, classes):
+        if old_name in self.results:
+            del self.results[old_name]
+        if new_name == old_name:
+            self.cls[old_name] = classes
+            self.update_cls_file()
+            return old_name
+        else:
+            del self.cls[old_name]
+            active = self.active[old_name]
+            del self.active[old_name]
+            return self.insert_new_cls(new_name, classes, active)
+
+    def delete_cls(self, id):
+        if id in self.results:
+            del self.results[id]
+        del self.cls[id]
+        del self.active[id]
+        self.update_cls_file()
+
+    def update_cls_file(self):
+        with open(self.cls_file_path, 'wb') as file:
+            pickle.dump(self.cls, file, protocol=pickle.HIGHEST_PROTOCOL)
+
     def get_bounding_box_fig(self, cls_id, img: Image):
-        plot = self.analyze_or_from_cache(cls_id, img).plot()
-        hight, length = np.shape(plot)[0:2]
-        scale = 10/min(length, hight)
-        with warnings.catch_warnings(action="ignore"):
-            fig = plt.figure(frameon=False, figsize=(length*scale,hight*scale))
-            ax = fig.add_axes([0, 0, 1, 1])
-            ax.imshow(plot)
-        return fig
-
-    def get_blurred_as_list(self, cls_ids: str, img: Image):
+        bb_image = merge_results_list(self.analyze_or_from_cache(cls_id, img)).plot()
+        image = Image.fromarray(bb_image)
```

Comment:
The class `Image` refers to the dataclass and not to `PIL.Image`.

```suggestion
        image = PIL.Image.fromarray(bb_image)
```